### PR TITLE
Added keyboard support for trigger and shoulder buttons

### PR DESCRIPTION
**Pull request by ObsidianGuardian**

### CODE REVIEW

**Comment by maxton** [comment]
It appears you're using tabs instead of two spaces. You need to change that to conform to [the style guide][1]. Following that, be sure to squash your commits to one.

[1]: docs/style_guide.md

File: src/xenia/hid/winkey/winkey_input_driver.cc [hunk 1]
```diff
@@ -304,6 +312,22 @@ X_RESULT WinKeyInputDriver::GetKeystroke(uint32_t user_index, uint32_t flags,
     // Z
     virtual_key = 0x5815;  // VK_PAD_BACK
   }
+  if (evt.vkey == (0x51) || 0x49) {
```
Comment (DrChat) [inline src/xenia/hid/winkey/winkey_input_driver.cc]: This check will always pass. Change `|| 0x49` to `|| evt.vkey == 0x49`

File: src/xenia/hid/winkey/winkey_input_driver.cc [hunk 2]
```diff
@@ -304,6 +312,22 @@ X_RESULT WinKeyInputDriver::GetKeystroke(uint32_t user_index, uint32_t flags,
...
+	//Q / I
+	virtual_key = 0x5806;	// VK_PAD_LTRIGGER
+  }
+  if (evt.vkey == (0x45) || 0x4F) {
```
Comment (DrChat) [inline src/xenia/hid/winkey/winkey_input_driver.cc]: Ditto.

**Comment by DrChat** [comment]
Good changes!

As @maxton said, get this code passing clang-format (if you're using visual studio, I'd highly suggest [this plugin](http://llvm.org/builds/) (at the bottom of the page), and then squash all of these commits into one ([e.g. like this](http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git)).

**Comment by ObsidianGuardian** [comment]
Made the  requested changed, fixed my formatting and squashed everything into one commit.

**Comment by Deathlock47** [comment]
Weren't triggers already bound to q and e?

**Comment by ObsidianGuardian** [comment]
triggers were bound but were not mapped to the xinput(controller) keys

**Comment by DrChat** [comment]
Cool beans - one last request: Could you rewrite that commit message to just the first line?

File: src/xenia/hid/winkey/winkey_input_driver.cc [hunk 3]
```diff
@@ -304,6 +312,22 @@ X_RESULT WinKeyInputDriver::GetKeystroke(uint32_t user_index, uint32_t flags,
...
+  }
+  if (evt.vkey == (0x45) || evt.vkey == (0x4F)) {
+    // E / O
+    virtual_key = 0x5807;  // VK_PAD_RTRIGGER
+  }
+  if (evt.vkey == (0x31)) {
+    // 1
+    virtual_key = 0x5805;  // VK_PAD_LSHOULDER
+  }
+  if (evt.vkey = (0x33)) {
```
Comment (tgsm) [inline src/xenia/hid/winkey/winkey_input_driver.cc]: Should be `==`

Comment (paulsapps) [inline src/xenia/hid/winkey/winkey_input_driver.cc]: could evt be changed to const to prevent this? :)

Comment (ObsidianGuardian) [inline src/xenia/hid/winkey/winkey_input_driver.cc]: @tgsm True changed it in the newest commit.

**Comment by DrChat** [comment]
LGTM 👍

**Pull request merged**